### PR TITLE
build(ci): ensure `testlib` is built

### LINKED ISSUE #17520: CI: ensure `testlib` is built
See 

* https://github.com/ankidroid/Anki-Android/issues/17517

We want to ensure that GitHub Actions runs `./gradlew :testlib:assemblePlayDebugAndroidTest` at some point to ensure that `testlib` compiles

This could likely either be done directly in CI (`/.github`), or by adding a dependency on this task from AnkiDroid's `androidTest` in `build.gradle`. The latter being preferred

### PR DESCRIPTION
**Pull request by tanishy7777**
Added a dependency on the assemblePlayDebugAndroidTest task from AnkiDroid's androidTest in build.gradle.

## Purpose / Description
Fixes the issue #17520 by adding  a dependency on the assemblePlayDebugAndroidTest task from AnkiDroid's androidTest in build.gradle.

## Fixes
* Fixes #17520 

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/build.gradle [hunk 1]
```diff
@@ -281,6 +281,11 @@ tasks.register('copyTestLibIntoAndroidTest', Copy) {
 tasks.named('preBuild').configure { dependsOn('copyTestLibIntoAndroidTest') }
 tasks.named('runKtlintCheckOverAndroidTestSourceSet').configure { mustRunAfter('copyTestLibIntoAndroidTest') }
 
+// to run manually: `./gradlew :testlib:assemblePlayDebugAndroidTest`
+tasks.named("assemblePlayDebugAndroidTest") {
```
Comment (mikehardy) [inline AnkiDroid/build.gradle:285]: The other task dependencies use `.configure` for a reason: it is done that way in order to work well with Gradle's ability to defer configuration, which can speed compilation for jobs that don't use some tasks and thus don't need to configure them

This was suggested as something to change via gradle lint rules in one of the recent gradle updates

https://github.com/ankidroid/Anki-Android/pull/14855/commits/68b1f68a9dd3950bd2a633af89175daea915a6e3

```suggestion
tasks.named("assemblePlayDebugAndroidTest").configure {
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hey there 👋  welcome to AnkiDroid

One subtle suggestion here but otherwise, this seems like it should work!

You erased the testing section of the PR template though so I am not sure how you proved to yourself it worked? 

Please re-insert that section, then show the results of a `./gradlew androidTest -v` (I think `-v` prints out the executed tasks...) before and after the change to see if the new task dependency is definitely missing before and definitely there after in the list of executed tasks

**Comment by mikehardy** [comment]
Actually, it looks like it didn't compile at all

> 
> * Where:
> Build file '/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/build.gradle' line: 285

Did you run this prior to submitting it?

**Comment by MritunjayTiwari14** [comment]
Shall i work on this issue?

**Comment by mikehardy** [comment]
@tanishy7777 do you intend to get this working locally and respond to the comment?

@MritunjayTiwari14 I'm not sure what you are talking about. I'm not sure what hamcrest etc have to do with it, I don't see any errors related to that.

**Comment by MritunjayTiwari14** [comment]
Alright, just saw the other PR where David had already fixed the Dependency issue.

**Comment by david-allison** [comment]
I presume my PR was: 

* https://github.com/ankidroid/Anki-Android/pull/17519

To note: this issue (#17520) is a follow-on from that PR

* #17520

**Comment by MritunjayTiwari14** [comment]
@david-allison just want to confirm that is Gradle still throwing those duplicate class errors while trying to compile?

**Pull request closed without merging**